oauth2client/contrib/gce.py

```diff
@@ -51,13 +59,17 @@ class AppAssertionCredentials(AssertionCredentials):
     """
 
     @util.positional(2)
-    def __init__(self, scope, **kwargs):
+    def __init__(self, scope='', **kwargs):
         """Constructor for AppAssertionCredentials
 
         Args:
             scope: string or iterable of strings, scope(s) of the credentials
-                   being requested.
+                   being requested. Using this argument will have no effect on
+                   the actual scopes for tokens requested. These scopes are
+                   set at VM instance creation time and won't change.
         """
+        if scope:
+            warnings.warn(_SCOPES_WARNING)
         self.scope = util.scopes_to_string(scope)
```

Comment:
This class had no children, only parents. And as a child at the bottom of the hierarchy, it has no use for scope because of GCE (essentially what is in the warning message).

I can add a comment. How does this work:

```python
# This is just provided for backwards compatibility, but is not used by this class.
```